### PR TITLE
Group cities

### PR DESCRIPTION
**Pull request by Eschults**
ping @ssaunier 

capture d ecran 2016-09-05 a 14 31 40

### CODE REVIEW

**Comment by Eschults** [comment]
Having second thoughts about this one, should we keep `/cities` as it is and add a `/groups` endpoint @ssaunier ? Plz let me know before I udpate code in `www`

**Comment by ssaunier** [comment]
You're right, `/city_groups` make more sens to follow REST principle on this new model.

File: db/migrate/20160905110400_add_group_reference_to_cities.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    # create Brazilian cities ordered list
+    brazil_ordered_list = OrderedList.create!(
+      name: "brazil_official_cities",
+      element_type: "City",
+      slugs: brazil_slugs
+    )
+  end
+
+end
```
Comment (ssaunier) [inline db/migrate/20160905110400_add_group_reference_to_cities.rb]: Love these two migrations 👌 !

File: db/migrate/20160905110400_add_group_reference_to_cities.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
+class AddGroupReferenceToCities < ActiveRecord::Migration
+  def change
+    add_reference :cities, :city_group, index: true, foreign_key: true
+    # associate French cities to france's city group
+    france_slugs = %w(paris bordeaux lille aix-marseille nantes)
```
Comment (ssaunier) [inline db/migrate/20160905110400_add_group_reference_to_cities.rb]: `lyon` is missing.

File: db/migrate/20160905110400_add_group_reference_to_cities.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    # create European cities ordered list
+    europe_ordered_list = OrderedList.create!(
+      name: "europe_official_cities",
+      element_type: "City",
+      slugs: europe_slugs
+    )
+
+    # associate Brazilian cities to brazil's city group
+    brazil_slugs = %w(sao-paulo belo-horizonte)
```
Comment (ssaunier) [inline db/migrate/20160905110400_add_group_reference_to_cities.rb]: `rio` is missing.

File: db/migrate/20160905110400_add_group_reference_to_cities.rb [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    france = CityGroup.find_by(slug: "france")
+    france_slugs.each do |slug|
+      city = City.find_by(slug: slug)
+      city.city_group = france
+      city.save!
+    end
+
+    # create French cities ordered list
+    france_ordered_list = OrderedList.create!(
+      name: "france_official_cities",
```
Comment (ssaunier) [inline db/migrate/20160905110400_add_group_reference_to_cities.rb]: Why `_official_` ? Why not just having `:city_goup_cities`?

File: lib/tasks/db.rake [hunk 5]
```diff
@@ -1,3 +1,21 @@
...
+  desc "Dump AWS production DB and restore it locally."
+  task import_from_aws: [ :environment, :create ] do
+    c = Rails.configuration.database_configuration[Rails.env]
+    Bundler.with_clean_env do
+      puts "[1/4] Fetching DB password from Heroku"
+      db_url = `heroku config:get DATABASE_URL`
+      db_password = URI(db_url).password
 
+      puts "[2/4] Dumping DB"
+      `PGPASSWORD=#{db_password} pg_dump -h kitt.cwtxaafignjt.eu-west-1.rds.amazonaws.com -U alumni -d alumni -F c -b -v -f tmp/alumni_prod_rds.dump`
```
Comment (ssaunier) [inline lib/tasks/db.rake]: Just realized that we could derive the `host` here from the `URI(db_url)`, would be more resilient to host change in the future.

File: app/models/city_group.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+#  id         :integer          not null, primary key
+#  name       :string
+#  slug       :string
+#  icon       :text
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+
+class CityGroup < ActiveRecord::Base
+  has_many :cities, dependent: :nullify
```
Comment (ssaunier) [inline app/models/city_group.rb]: :+1: for the `nullify` strategy 😅

File: lib/tasks/db.rake [hunk 7]
```diff
@@ -1,3 +1,21 @@
...
+  desc "Dump AWS production DB and restore it locally."
+  task import_from_aws: [ :environment, :create ] do
+    c = Rails.configuration.database_configuration[Rails.env]
+    Bundler.with_clean_env do
+      puts "[1/4] Fetching DB password from Heroku"
+      db_url = `heroku config:get DATABASE_URL`
+      db_password = URI(db_url).password
 
+      puts "[2/4] Dumping DB"
+      `PGPASSWORD=#{db_password} pg_dump -h #{db_url.to_s.scan(/@.*\.com/).first[1..-1]} -U alumni -d alumni -F c -b -v -f tmp/alumni_prod_rds.dump`
```
Comment (ssaunier) [inline lib/tasks/db.rake]: Actually you could have just used:

```ruby
URI(db_url).host
```

That module does all this heavy lifting. Pushing on the branch.

**Pull request merged**